src/randomizer.py

```diff
@@ -1429,10 +1433,18 @@ def visit_Assignment(self, node: ca.Assignment) -> None:
     assign = random.choice(cands)
     var = assign.lvalue
 
+    ins_cands = get_insertion_points(fn, region)
+    ensure(ins_cands)
+
+    for ins_block, ins_index, node in ins_cands:
+        if node is assign:
+            break
+    else:
+        assert False, "original assign should always be found in insertion points"
```

Comment:
either replace by `raise RandomizationFailure`, or move this down back again (since all ensures are gone, which are the only reason to keep this up here)